Source/CarthageKit/Xcode.swift

```diff
@@ -189,10 +189,24 @@ extension BuildArguments: CustomStringConvertible {
 public func locateProjectsInDirectory(directoryURL: NSURL) -> SignalProducer<ProjectLocator, CarthageError> {
 	let enumerationOptions: NSDirectoryEnumerationOptions = [ .SkipsHiddenFiles, .SkipsPackageDescendants ]
 
+	let submodules = submodulesInRepository(directoryURL)
+		.map { return $0.path }
+		.collect()
+		.first()?.value ?? []
```

Comment:
`.first()` is usually discouraged outside of tests. The idiomatic way would be to `collect()` and then do the next step inside a `flatMap`.